src/App.tsx

```diff
@@ -14,11 +16,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [isStarted, setIsStarted] = useState(false);
+  const [sort, setSort] = useState('');
+  const [lengthLimit, setLimit] = useState(0);
+  const wordLength = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+  const visibleGoods = [...goodsFromServer]
+    .filter(good => good.length >= lengthLimit);
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sort) {
+      case 'Alf':
+        return g1.localeCompare(g2);
+      case 'length':
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return (
+    <div className="App">
+      {!isStarted
+        ? (
+          <button
+            type="button"
+            className="button is-warning button--size"
+            onClick={() => setIsStarted(true)}
+          >
+            START
+          </button>
+        )
+        : undefined}
+      {isStarted && <GoodList goods={visibleGoods} />}
+
+      <div className="button__flex">
+        {isStarted && (
+          <button
+            className="button is-success"
+            type="button"
+            onClick={() => setIsReversed(true)}
+          >
+            REVERSE
+          </button>
+        )}
+        {isStarted && (
+          <button
+            className="button is-success"
+            type="button"
+            onClick={() => setSort('Alf')}
+          >
+            Sort alphabetically
+          </button>
+        )}
+        {isStarted && (
+          <button
+            className="button is-success"
+            type="button"
+            onClick={() => setSort('length')}
+          >
+            Sort by length
+          </button>
+        )}
+        {isStarted && (
+          <button
+            className="button is-success"
+            type="button"
+            onClick={() => {
+              return (
+                setSort(''),
+                setIsReversed(false),
+                setLimit(0)
+              );
+            }}
```

Comment:
you should not implement handler logic in place, that would be great if you create a separate function-handler and just pass it here

Reply:
and the same way all over your code